src/EditorFeatures/Core.Cocoa/Lsp/VSMacLspLoggerFactoryWrapper.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.VSMac.API;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.EditorFeatures.Cocoa.Lsp;
+
+/// <summary>
+/// Wraps the external access <see cref="IVSMacLspLoggerFactory"/> and exports it
+/// as an <see cref="ILspLoggerFactory"/> for inclusion in the vsmac composition.
+/// </summary>
+[Export(typeof(ILspLoggerFactory)), Shared]
+internal class VSMacLspLoggerFactoryWrapper : ILspLoggerFactory
```

Comment:
On the Windows side, this factory is in the VS layer, not the EditorFeatures.Wpf layer. Typically that means the implementation should go in VSMac itself. I know we've discussed the idea of starting to move VSMac's Roslyn integration from VSMac to this repo, but we're not there yet. Whatever the reason is for wanting to implement this in Roslyn instead of VSMac right now, maybe we need a new "VSMac layer" assembly here, so that the purpose of `EditorFeatures.Cocoa` doesn't diverge too much from `EditorFeatures.Wpf`.

Reply:
Technically there is no reason that the windows impl could not live in the editor features layer, other than at some point the host workspace kind may need to differ between VS and VSMac.  At some point in the future I am potentially thinking about unifying all of them as well around a single impl in EditorFeatures, so I think we don't need a new VSMac layer for this right now